rumble Error

```java
package com.youlu.core.gamepad;

import com.youlu.common.utils.MoveByUtil;
import com.youlu.core.gamepad.model.GamePadModel;
import com.youlu.core.sound.SoundPlayService;
import de.gurkenlabs.input4j.InputDevice;
import de.gurkenlabs.input4j.InputDevicePlugin;
import de.gurkenlabs.input4j.InputDevices;
import jakarta.annotation.PostConstruct;
import jakarta.annotation.PreDestroy;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.Collection;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.TimeUnit;
import java.util.function.BiConsumer;
import java.util.function.Function;

/**
 * @author lsl
 * @version 1.0
 * @since 2025.12.26 19:57
 */
@Slf4j
@Data
@Component
@RequiredArgsConstructor
public class GamePadService {

    private volatile boolean flag = true;

    private boolean lastLeftShoulder = false;
    private boolean lastRightShoulder = false;
    private boolean xButton = false;

    private final SoundPlayService soundPlayService;

    @Qualifier("gamepadThreadPool")
    private final ExecutorService executorService;

    @PostConstruct
    public void init() {
        CompletableFuture.runAsync(() -> {
                    try (InputDevicePlugin inputDevices = InputDevices.init()) {
                        Collection<InputDevice> inputDeviceCollection = inputDevices.getAll();
                        log.info("gamepad size: {}", inputDeviceCollection.size());
                        if (!inputDeviceCollection.isEmpty()) {
                            soundPlayService.playWav("voices/gamepad_connected.wav");
                        }
                        while (!inputDevices.getAll().isEmpty() && flag) {
                            for (var inputDevice : inputDevices.getAll()) {
                                inputDevice.poll();
                                GamePadModel gamePadModel = new GamePadModel();
                                gamePadModel.setName(inputDevice.getName());
                                gamePadModel.setDpadUp(inputDevice.getComponents().getFirst().getData() == 1);
                                gamePadModel.setDpadDown(inputDevice.getComponents().get(1).getData() == 1);
                                gamePadModel.setDpadLeft(inputDevice.getComponents().get(2).getData() == 1);
                                gamePadModel.setDpadRight(inputDevice.getComponents().get(3).getData() == 1);
                                gamePadModel.setStart(inputDevice.getComponents().get(4).getData() == 1);
                                gamePadModel.setBack(inputDevice.getComponents().get(5).getData() == 1);
                                gamePadModel.setLeftThumb(inputDevice.getComponents().get(6).getData() == 1);
                                gamePadModel.setRightThumb(inputDevice.getComponents().get(7).getData() == 1);
                                gamePadModel.setLeftShoulder(inputDevice.getComponents().get(8).getData() == 1);
                                gamePadModel.setRightShoulder(inputDevice.getComponents().get(9).getData() == 1);
                                gamePadModel.setA(inputDevice.getComponents().get(10).getData() == 1);
                                gamePadModel.setB(inputDevice.getComponents().get(11).getData() == 1);
                                gamePadModel.setX(inputDevice.getComponents().get(12).getData() == 1);
                                gamePadModel.setY(inputDevice.getComponents().get(13).getData() == 1);
                                gamePadModel.setLeftTrigger(inputDevice.getComponents().get(14).getData());
                                gamePadModel.setRightTrigger(inputDevice.getComponents().get(15).getData());
                                gamePadModel.setLeftThumbX(inputDevice.getComponents().get(16).getData());
                                gamePadModel.setLeftThumbY(inputDevice.getComponents().get(17).getData());
                                gamePadModel.setRightThumbX(inputDevice.getComponents().get(18).getData());
                                gamePadModel.setRightThumbY(inputDevice.getComponents().get(19).getData());

                                sendMessage(gamePadModel, inputDevice);
                            }

                            TimeUnit.MILLISECONDS.sleep(70);
                        }
                    } catch (IOException | InterruptedException e) {
                        throw new RuntimeException(e.getMessage(), e);
                    }
                }, executorService)
                .whenComplete((_, exception) -> {
                    flag = false;
                    if (exception != null) {
                        log.error("gamepad error", exception);
                    }
                });
    }

    @PreDestroy
    public void destroy() {
        flag = false;
    }

    public void sendMessage(GamePadModel gamePadModel, InputDevice inputDevice) {
        float rightThumbX = gamePadModel.getRightThumbX();
        float leftThumbY = gamePadModel.getLeftThumbY();
        float rightTrigger = gamePadModel.getRightTrigger();

        boolean currentLeftShoulder = gamePadModel.isLeftShoulder();
        boolean currentRightShoulder = gamePadModel.isRightShoulder();
        boolean currentXButton = gamePadModel.isX();

        if (currentLeftShoulder && !lastLeftShoulder) {
        }

        if (currentRightShoulder && !lastRightShoulder) {
        }

        if (currentXButton && !xButton) {
            soundPlayService.playWav("voices/honk.wav", false);

            inputDevice.rumble(new float[]{0.8f, 0.3f});
        }

        lastLeftShoulder = currentLeftShoulder;
        lastRightShoulder = currentRightShoulder;
        xButton = currentXButton;

        float[] filteredLeft = MoveByUtil.filterJoystick(rightThumbX, leftThumbY);
        float filteredLeftX = filteredLeft[0];
        float filteredLeftY = filteredLeft[1];

        float currentLeftThumbY = filteredLeftY * (0.2f + 0.8f * rightTrigger);

        if (filteredLeftX != 0 || currentLeftThumbY != 0) {
            float linearVelocity = MoveByUtil.buildLinearVelocity(currentLeftThumbY);
            float angularVelocity = MoveByUtil.buildAngularVelocity(filteredLeftX);
            if (linearVelocity != 0 || angularVelocity != 0) {
                log.info("线linearVelocity: {}, angularVelocity: {}", linearVelocity, angularVelocity);
            }
        }
    }

}

````

The above is my test code. Calling Rumble on a Windows 11 23h2 system results in an error.

```text
Error:
Caused by: java.lang.NumberFormatException: For input string: "Gamepad (0)"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67) ~[?:?]
	at java.base/java.lang.Integer.parseInt(Integer.java:565) ~[?:?]
	at java.base/java.lang.Integer.parseInt(Integer.java:662) ~[?:?]
	at de.gurkenlabs.input4j.foreign.windows.xinput.XInputPlugin.rumbleXInputDevice(XInputPlugin.java:197) ~[input4j-1.1.1.jar:?]
	at de.gurkenlabs.input4j.InputDevice.rumble(InputDevice.java:247) ~[input4j-1.1.1.jar:?]
	at com.youlu.core.gamepad.GamePadService.sendMessage(GamePadService.java:145) ~[classes/:?]
```

--- a/src/main/java/de/gurkenlabs/input4j/foreign/windows/xinput/XInputPlugin.java
+++ b/src/main/java/de/gurkenlabs/input4j/foreign/windows/xinput/XInputPlugin.java
@@ -194,9 +194,20 @@ public final class XInputPlugin extends AbstractInputDevicePlugin {
     }
 
     // Set the vibration for each motor (example for two motors)
-    setVibration(Integer.parseInt(inputDevice.getName()),
+    setVibration(resolveDeviceId(inputDevice),
       (short) (motorSpeedLeft * XINPUT_VIBRATION.MAX_VIBRATION),
       (short) (motorSpeedRight * XINPUT_VIBRATION.MAX_VIBRATION));
+  }
+
+  /**
+   * Resolves the numeric device ID from the input device's identifier.
+   *
+   * @param inputDevice The input device.
+   * @return The numeric device ID.
+   * @throws NumberFormatException If the device identifier is not a valid integer.
+   */
+  static int resolveDeviceId(InputDevice inputDevice) {
+    return Integer.parseInt(inputDevice.getID());
   }
 
   /**
@@ -209,8 +220,7 @@ public final class XInputPlugin extends AbstractInputDevicePlugin {
     this.refreshDevices();
     var polledValues = new float[inputDevice.getComponents().size()];
 
-    var deviceId = Integer.parseInt(inputDevice.getID());
-    var state = getState(deviceId);
+    var state = getState(resolveDeviceId(inputDevice));
     if (state == null) {
       return new float[0];
     }

--- a/src/test/java/de/gurkenlabs/input4j/foreign/windows/xinput/XInputPluginTests.java
+++ b/src/test/java/de/gurkenlabs/input4j/foreign/windows/xinput/XInputPluginTests.java
@@ -1,11 +1,13 @@
 package de.gurkenlabs.input4j.foreign.windows.xinput;
 
+import de.gurkenlabs.input4j.InputDevice;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.condition.EnabledOnOs;
 import org.junit.jupiter.api.condition.OS;
 
 import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class XInputPluginTests {
   @Test
@@ -61,5 +63,22 @@ public class XInputPluginTests {
 
     // Test case 7: Negative value outside deadzone
     assertEquals(-0.5f, XInputPlugin.normalizeSignedShort((short) -16384, 1000), 0.01);
+  }
+
+  @Test
+  @EnabledOnOs(OS.WINDOWS)
+  void testResolveDeviceId() {
+    var device = new InputDevice("0", "Gamepad (0)", null, _ -> new float[]{}, (_, _) -> {});
+    assertEquals(0, XInputPlugin.resolveDeviceId(device));
+
+    var device2 = new InputDevice("3", "XInput Device (3)", null, _ -> new float[]{}, (_, _) -> {});
+    assertEquals(3, XInputPlugin.resolveDeviceId(device2));
+  }
+
+  @Test
+  @EnabledOnOs(OS.WINDOWS)
+  void testResolveDeviceIdThrowsForNonNumericName() {
+    var device = new InputDevice("Gamepad (0)", "Gamepad (0)", null, _ -> new float[]{}, (_, _) -> {});
+    assertThrows(NumberFormatException.class, () -> XInputPlugin.resolveDeviceId(device));
   }
 }

